### ISSUE TITLE
Feature: Balanced Accuracy

### ISSUE THREAD

**Issue by jondo** [tensorflow/tensorflow#57615]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Feature Request

### Source

binary

### Tensorflow Version

tf 2.10.0-rc3

### Custom Code

No

### OS Platform and Distribution

Debian 11

### Python version

3.9

### CUDA/cuDNN version

11.2

### Current Behaviour?

I suggest adding [Balanced Accuracy](https://www.statology.org/balanced-accuracy/) as additional [Keras metrics](https://keras.io/api/metrics/).

It is useful for training with an imbalanced validation dataset.

(Note: I see various suggestions on StackOverflow: [1](https://stackoverflow.com/questions/59339531/balanced-accuracy-score-in-tensorflow), [2](https://stackoverflow.com/questions/65189262/tensorflow-2-0-custom-metric-balanced-accuracy-score-for-modelcheckpoint-not))

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
metrics = [tf.keras.metrics.BalancedAccuracy()]
```

### Relevant log output

_No response_</details>

**Comment by tilakrayal** [comment]
@jondo,
Could you please try to use **tfma.metrics.BalancedAccuracy** as mentioned in this doc [link](https://www.tensorflow.org/tfx/model_analysis/api_docs/python/tfma/metrics/BalancedAccuracy) & [SO link](https://stackoverflow.com/questions/59339531/balanced-accuracy-score-in-tensorflow) and let us know if you are looking for the similar feature. Thank you!

**Comment by EliaSchiavon** [comment]
I think that the request regards the implementation of the BalancedAccuracy metric as an option in [tf.keras.metrics](https://www.tensorflow.org/api_docs/python/tf/keras/metrics). Actually, the proposed method in the SO link is to write a custom metric class, while the documentation linked is referring to TFX and not to the standard TF library

**Comment by Mustafa-MS** [comment]
The BalancedAccuracy metric needs to be implemented in the standard TF library, It's really helpful with imbalanced data.

**Comment by MSzele** [comment]
Totally agree with @Mustafa-MS, any updates on this?
